aggregators/internal/hdrhistogram/hdrhistogram.go

```diff
@@ -278,3 +277,90 @@ func getBucketCount() int32 {
 	}
 	return bucketsNeeded
 }
+
+// HybridCountsRep represents a hybrid counts representation for
+// sparse histogram. It is optimized to record a single value as
+// integer type and more values as map.
+type HybridCountsRep struct {
+	bucket int32
+	value  int64
+	m      map[int32]int64
+}
+
+// Add adds a new value to a bucket of given index.
+func (c *HybridCountsRep) Add(bucket int32, value int64) {
+	if c.m == nil && c.bucket == 0 && c.value == 0 {
```

Comment:
Should we be checking `len(c.m) == 0` rather than `c.m == nil`, in case the map has been reset/cleared? Or I suppose it doesn't matter - if the map has already been allocated the cost is already paid?

Reply:
> Or I suppose it doesn't matter - if the map has already been allocated the cost is already paid?

This was my reasoning for not checking `len` here. Adding entries will also allocate but since this is a matter of a single entry in a map it should not be extra allocations. Does that make sense?

Reply:
Yep, I think it's fine - let's leave it as it is.